content/concepts/did-ddo.md

```diff
@@ -369,7 +372,100 @@ Example:
   ]
 }
 ```
+ 
+#### User Input
+
+Sometimes, you may need some input before consuming a dataset or running an algorithm.
+Examples:
+
+- You want to know the desired sampling interval of data in your dataset, before the user is going to download it. Your dataset URL is 'www.example.com/mydata'. So you will define a field called "sampling", ask the user to enter a value and then this parameter is going to be added to the URL of your dataset (so the actual url will become 'www.example.com/mydata?sampling=10'
+- Before running an algorithm, you need to know how many iterations should the algo perform.  You are going to define a field called 'iterations', ask the user to enter a value and this parameter is stored in a specific location in your C2D pod, so algo can read it and use that value.
+
+Object consists of an array , in which each element defines a user required field.
+An element looks like:
+
+```json
+   {
+        "name":"surname",
+        "type": "text",
+        "label": "Name",
+        "required": true,
+        "description":"Please fill your name",
+        "default": "NoName",
+        "options": []
+    }
+```
 
+where:
+
+  - name  = defines the parameter name (this is sent as HTTP param or key towards algo)
+  - type  = defines the form type  (text, number, select, boolean)
+  - label = defines the label which is displayed
+  - required = if this field is mandatory to have a user input.
+  - default  = default value
+  - description = description of this element
+  - options = for select types, a list of options
+
+
+Example:
+
+```json
+[
+   {
+       "name":"surname",
+        "type": "text",
+        "label": "Name",
+        "required": true,
+        "default": "NoName"
+       "description":"Please fill your name"
+    },
+   {
+       "name":"age",
+        "type": "number",
+        "label": "Age",
+        "required": false,
+        "default": 0
+        "description":"Please fill your age"
+    },
+   {
+       "name":"developer",
+        "type": "boolean",
+        "label": "Developer",
+        "required": false,
+        "default": false
+        "description":"Are you a developer?"
+    },
+    {
+       "name":"preference",
+        "type": "select",
+        "label": "Date",
+        "required": false,
+        "default": "nodejs"
+        "options": [
+             {
+                    "nodejs" : "I love NodeJs"
+             }, 
+             {
+                  "python" : "I love Python"
+             }
+        ],
+        "description": "Do you like NodeJs or Python"
+    },
+]
+```
+
+Algorithms will have access to a json file located at /data/inputs/algoCustomData.json, which contains the keys/values for input data required.  Example:
```

Comment:
i don't really understand this.

Reply:
this will be also documented in C2D docs.  It's about from where the algorithm can read those values.